php/class-pwawp-app-manifest.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * PWAWP_APP_Manifest class.
+ *
+ * @package PWA
+ */
+
+/**
+ * PWAWP_APP_Manifest class.
+ *
+ * Mainly copied from Jetpack_PWA_Manifest and Jetpack_PWA_Helpers.
+ */
+class PWAWP_APP_Manifest {
+
+	/**
+	 * The query arg that is present in a request for the app manifest.
+	 *
+	 * @var string
+	 */
+	const MANIFEST_QUERY_ARG = 'pwawp_manifest';
+
+	/**
+	 * The theme color to use if no dynamic values are present.
+	 *
+	 * @var string
+	 */
+	const FALLBACK_THEME_COLOR = '#fff';
+
+	/**
+	 * The default manifest icon sizes.
+	 *
+	 * Copied from Jetpack_PWA_Helpers::get_default_manifest_icon_sizes().
+	 * Based on a conversation in https://github.com/GoogleChrome/lighthouse/issues/291
+	 *
+	 * @var int[]
+	 */
+	public $default_manifest_icon_sizes = array( 192, 512 );
+
+	/**
+	 * Inits the class.
+	 *
+	 * Mainly copied from Jetpack_PWA_Manifest::__construct().
+	 */
+	public function init() {
+		add_action( 'wp_head', array( $this, 'manifest_link_and_meta' ) );
+		add_action( 'amp_post_template_head', array( $this, 'manifest_link_and_meta' ) );
+		add_action( 'template_redirect', array( $this, 'send_manifest_json' ), 2 );
+	}
+
+	/**
+	 * Outputs the <link> and <meta> tags for the app manifest.
+	 *
+	 * Mainly copied from Jetpack_PWA_Manifest::render_manifest_link().
+	 */
+	public function manifest_link_and_meta() {
+		?>
+		<link rel="manifest" href="<?php echo esc_url( add_query_arg( self::MANIFEST_QUERY_ARG, '1', home_url( '/' ) ) ); ?>">
```

Comment:
Note below about using REST API for the URL

Reply:
[3bc37](https://github.com/xwp/pwa-wp/pull/13/commits/3bc378a31ce34a6bc0621ed9e55c71ae9e4421ce) adds a REST API endpoint for the manifest, and uses it in the `<link rel="manifest">`.